addons/website_profile/views/website_profile.xml

```diff
@@ -326,25 +329,16 @@
                 </linearGradient>
             </svg>
             <div class="o_pc_overlay d-flex flex-column align-items-center justify-content-center">
-                <img class="img-fluid"
-                    t-att-src="website.image_url(next_rank_id if next_rank_id else user.rank_id, 'image_128')"
-                    t-att-alt="(next_rank_id.name if next_rank_id else user.rank_id.name) + ' badge'"
-                    t-att-style="'max-width: ' + (img_max_width if img_max_width else '50%;')"/>
-                <h4 class=" mb-0">
-                    <span t-if="next_rank_id" t-field="next_rank_id.name"/>
-                    <span t-else="" t-field="user.rank_id.name"/>
-                </h4>
-                <small>
-                    <span class="fw-bold text-primary"
-                        t-field="user.karma"
-                        t-options="{'format_decimalized_number': True, 'precision_digits': 2}"/> /
-                    <span t-if="next_rank_id" class="fw-bold"
-                        t-field="next_rank_id.karma_min"
-                        t-options="{'format_decimalized_number': True, 'precision_digits': 2}"/>
-                    <span t-else="" class="fw-bold"
-                        t-field="user.rank_id.karma_min"
-                        t-options="{'format_decimalized_number': True, 'precision_digits': 2}"/>
-                     xp
+                <t t-set="rank_id" t-value="next_rank_id if next_rank_id else user.rank_id"/>
+                <div t-field="rank_id.image_1920"
+                    t-attf-style="max-width: {{img_max_width or '50%'}};"
+                    t-options="{'widget': 'image', 'preview_image': 'image_128'}"/>
+                <h4 class="o_pc_text_overflow text-center" t-field="rank_id.name"/>
+                <small class="w-50 text-center">
+                    <t t-set="karma_value" t-value="next_rank_id.karma_min - user.karma if next_rank_id else user.karma"/>
+                    <span t-if="next_rank_id">Get</span>
```

Comment:
Would split this into 2 cases
t-if="next_rank_id"
...
t-else
...

It's currently a bit hard to follow.